crates/ruff_python_ast/src/comparable.rs

```diff
@@ -913,11 +910,31 @@ impl<'a> From<&'a ast::Expr> for ComparableExpr<'a> {
             }) => Self::FString(ExprFString {
                 values: values.iter().map(Into::into).collect(),
             }),
-            ast::Expr::Constant(ast::ExprConstant { value, range: _ }) => {
-                Self::Constant(ExprConstant {
+            ast::Expr::StringLiteral(ast::ExprStringLiteral {
+                value,
+                // Compare strings based on resolved value, not representation (i.e., ignore whether
+                // the string was implicitly concatenated).
+                implicit_concatenated: _,
+                unicode,
```

Comment:
The parser doesn't do anything except that to store the information in the AST node. I think it makes sense to ignore it here but I remember that I made a comment about this in some old PR. Let me try to find it, otherwise I'll remove it 😅

Reply:
Ah, https://github.com/astral-sh/ruff/pull/7180#discussion_r1316841244.

\cc @charliermarsh who might know more about this.